cmd/podman/cp.go

```diff
@@ -253,11 +272,58 @@ func copy(src, destPath, dest string, idMappingOpts storage.IDMappingOptions, ch
 	if srcfi.IsDir() {
 
 		logrus.Debugf("copying %q to %q", srcPath+string(os.PathSeparator)+"*", dest+string(os.PathSeparator)+"*")
-		if err = copyWithTar(srcPath, destPath); err != nil {
-			return errors.Wrapf(err, "error copying %q to %q", srcPath, dest)
+		if len(copyOpts.excludes) == 0 {
+			if err = copyWithTar(srcPath, destPath); err != nil {
+				return errors.Wrapf(err, "error copying %q to %q", srcPath, dest)
+			}
+			return nil
+		}
+		err := filepath.Walk(srcAbsPath, func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if info.IsDir() {
+				return nil
```

Comment:
what happens if a directory is marked to be excluded?

For example, if I've:
```
mkdir -p a/b
touch a/b/c

echo a/b > .dockerignore
```

Shouldn't we skip also the file `c` from being copied?

Reply:
fixed